### PR TITLE
qb: Improve various tests.

### PR DESCRIPTION
**Pull request by orbea**
## Description

This improves various qb configure tests and related issues.

## Related Issues

* Continues general refactorization for `config.libs.sh` with `HAVE_NETWORKING` to ensure these tests are more robust.

* Improves the various builtin library qb tests to be more robust and cleaner. We have four builtin libraries (`flac`, `mbedtls`, `miniupnpc` and `zlib`) and they are all enabled by default. They can be disabled with `--disable-builtinfoo` (.ie. `--disable-builtinflac`) or the entire feature including the system library can be disabled with `--disable-foo` (i.e. `--disable-flac`).

* Reworks the `BUILTIN_ZLIB` check in `Makefile.common` to match the other builtin checks better and to generally be less error prone.

* Removes a redundant `-DHAVE_COMPRESSION` define.

* Fixes a compile failure with `--disable-networking`.
```
command.c:2798:12: error: duplicate case value 'CMD_EVENT_NETPLAY_GAME_WATCH'
      case CMD_EVENT_NETPLAY_GAME_WATCH:
           ^
command.c:1762:12: note: previous case defined here
      case CMD_EVENT_NETPLAY_GAME_WATCH:
           ^
1 error generated.
```

### CODE REVIEW

File: qb/config.params.sh [hunk 1]
```diff
@@ -18,7 +18,7 @@ HAVE_STRIPES=no            # Stripes menu
 HAVE_RUNAHEAD=yes          # Runahead support
 HAVE_SSL=auto              # SSL/mbedtls support
 C89_SSL=no
-HAVE_BUILTINMBEDTLS=yes    # Bake in the mbedtls library
+HAVE_BUILTINMBEDTLS=auto   # Bake in the mbedtls library
```
Comment (twinaphex) [inline qb/config.params.sh]: Why are you doing this? We don't want to rely on system libs for this, I'd prefer to just set it to yes instead.

File: qb/config.params.sh [hunk 2]
```diff
@@ -45,7 +45,7 @@ HAVE_NETWORKGAMEPAD=auto   # Networked game pad (plus baked-in core)
 C89_NETWORKGAMEPAD=no
 HAVE_NETPLAYDISCOVERY=yes  # Add netplay discovery (room creation, etc.)
 HAVE_MINIUPNPC=auto        # Mini UPnP client library (for NAT traversal)
-HAVE_BUILTINMINIUPNPC=yes  # Bake in Mini UPnP client library (for NAT traversal)
+HAVE_BUILTINMINIUPNPC=auto # Bake in Mini UPnP client library (for NAT traversal)
```
Comment (twinaphex) [inline qb/config.params.sh]: Same here. We don't want to rely on system libs for this.

File: qb/config.params.sh [hunk 3]
```diff
@@ -114,7 +114,7 @@ HAVE_CHD=yes               # Compile in chd support
 CXX_CHD=no
 HAVE_7ZIP=yes              # Compile in 7z support
 HAVE_FLAC=auto             # Compile in flac support
-HAVE_BUILTINFLAC=yes       # Bake in flac support
+HAVE_BUILTINFLAC=auto      # Bake in flac support
```
Comment (twinaphex) [inline qb/config.params.sh]: Same here too. No system lib dependency for these things please by default.

**Comment by orbea** [comment]
@twinaphex The system library will only be preferred if they are set to `no` and in practice setting them to `auto` or `yes` will have no difference after this commit. I changed them to `auto` to reflect this and how they are more dynamic now.

So this will have no change in that regard and the builtin library for all four is the default.

I made sure the `check_enabled` calls preceded the relevant `check_lib` or etc. calls so they only are only aware of the user's configure arguments and not the configure tests for the system library.

**Comment by twinaphex** [comment]
OK, that seems OK then.

**Pull request merged**